azure_ai_search_rag.ipynb

```diff
@@ -0,0 +1,908 @@
+    "    \"arxiv_id\": x[\"arxiv_id\"],\n",
+    "})\n",
+    "\n",
+    "data = data.remove_columns([\n",
+    "    \"prechunk_id\", \"postchunk_id\", \"references\"\n",
+    "])\n",
+    "\n",
+    "data\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We need to define an embedding model to create our embedding vectors for retrieval, for that we will be using Mistral AI's `mistral-embed`. There is some cost associated with this model, so be aware of that (costs for running this notebook are <$1).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from mistralai.client import MistralClient\n",
+    "import getpass  # console.mistral.ai/api-keys/\n",
+    "\n",
+    "mistral_api_key = os.getenv(\"MISTRAL_API_KEY\") or getpass.getpass(\"Enter your Mistral API key: \")\n",
+    "\n",
+    "mistral = MistralClient(api_key=mistral_api_key)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embed_model = \"mistral-embed\"\n",
+    "\n",
+    "embeds = mistral.embeddings(\n",
+    "    model=embed_model, input=[\"this is a test\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can view the dimensionality of our returned embeddings, which we'll need soon when initializing our vector index:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1024"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dims = len(embeds.data[0].embedding)\n",
+    "dims"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Index Embeddings into Azure AI Search\n",
+    "\n",
+    "Now we create our vector DB to store our vectors. For this, we need to set up an [Azure AI Search service](https://portal.azure.com/#create/Microsoft.Search).\n",
+    "\n",
+    "There are two ways to authenticate to Azure AI Search:\n",
+    "\n",
+    "1. **Service Key**: The service key can be found in the \"Settings -> Keys\" section in the left navbar of the Azure portal dashboard. Make sure to select the ADMIN key.\n",
+    "2. **Managed Identity**: This is a more secure and recommended way to authenticate. You can follow the instructions in the [official Microsoft documentation](https://learn.microsoft.com/en-us/azure/search/search-howto-managed-identities-data-sources?tabs=portal-sys%2Cportal-user) to set up Managed Identity.\n",
+    "\n",
+    "For more detailed instructions on creating an Azure AI Search service, please refer to the [official Microsoft documentation](https://learn.microsoft.com/en-us/azure/search/search-create-service-portal).\n"
```

Comment:
Nit: it's better if you remove the language/locale from docs URLs so it goes to the users' preference, e.g. https://learn.microsoft.com/azure/search/search-create-service-portal for this one